package/MDAnalysis/lib/include/calc_distances.h

```diff
@@ -356,6 +356,59 @@ static void _triclinic_pbc(coordinate* coords, int numcoords, float* box)
     }
 }
 
+static void _translate_periodic_ortho(float* reference, float* centre, float* result, float* box)
+{  /*
+    Moves result to the periodic image of centre which is
+    closest to the reference point in orthorhombic unit cell.
+    */
+
+  double dx[3];
+  float inverse_box[3];
+
+  inverse_box[0] = 1.0 / box[0];
+  inverse_box[1] = 1.0 / box[1];
+  inverse_box[2] = 1.0 / box[2];
+
+  dx[0] = reference[0] - centre[0];
+  dx[1] = reference[1] - centre[1];
+  dx[2] = reference[2] - centre[2];
+
+  minimum_image(dx, box, inverse_box);
+
+  for (int i=0; i<3; i++) {
+      *(result+i)= reference[i] - dx[i];
```

Comment:
With respect to the `arrange_closest()` method, aren't we doing double work here? I mean here we compute a distance vector which we use to translate a coordinate. Back in `arrange_closest()`, we take this coordinate to compute the distance vector again... or am I getting something wrong here?